### PR TITLE
Add ingressAnnotations. Improve other syntaxes.

### PR DESCRIPTION
**Pull request by danielhoherd**
Related to <https://github.com/astronomer/issues/issues/2640>, which talks about GCP Global Access, and offers a solution of allowing arbitrary ingress annotations.

# Description

* Add default null string for loadBalancerIP
* Add default null dict for loadBalancerSourceRanges
* Fix yaml anchor syntax for platformTolerations
* Whitespace
* Add test for these changes and other existing behaviors of the NGINX Service
* Auto-formatting applied by vscode `prettier` that removed some whitespace, re-aligned some braces.

### CODE REVIEW

**Comment by danielhoherd** [comment]
May be related to https://github.com/astronomer/issues/issues/2700

File: charts/nginx/templates/nginx-service.yaml [hunk 1]
```diff
@@ -13,6 +13,11 @@ metadata:
     heritage: {{ .Release.Service }}
   annotations:
     service.beta.kubernetes.io/aws-load-balancer-type: "nlb"
```
Comment (danielhoherd) [inline charts/nginx/templates/nginx-service.yaml:15]: I feel like we should take this out and tell customers who need it to add it. This was very recently added.

Comment (danielhoherd) [reply charts/nginx/templates/nginx-service.yaml:15]: @ianstanton ^^^ ?

Comment (ianstanton) [reply charts/nginx/templates/nginx-service.yaml:15]: @danielhoherd I agree with this. Keeps things generic and gives customers the liberty to add this if they want. We'll need to add some instructions for this. @jwitz Who is in charge of document updates these days?

Comment (ianstanton) [inline charts/nginx/templates/nginx-service.yaml:15]:
```suggestion
```

**Review by ianstanton — CHANGES_REQUESTED** [review]

File: charts/nginx/values.yaml [hunk 2]
```diff
@@ -32,14 +32,17 @@ resources: {}
 #   cpu: 100m
 #   memory: 128Mi
 
+# dict of annotations to add to the ingress controller
```
Comment (rbankston) [inline charts/nginx/values.yaml:35]: Would like to make sure that we have an example either here in the helm chart or in documentation somewhere for customers that aren't as familiar with Go for what a dict looks like in the language to lessen questions to CRE or in the forum

File: charts/nginx/templates/nginx-service.yaml (hunk 1 above)
Comment (rbankston) [reply charts/nginx/templates/nginx-service.yaml:15]: what if service.beta.kubernetes.io/aws-load-balancer-type: "nlb" was removed from here and added a a default helm chart value as both documentation as code and to keep that default in place?

Comment (rbankston) [reply charts/nginx/templates/nginx-service.yaml:15]: Didn't like my other thought. Would like to see each of the deployment sections in the docs have common annotations for each cloud service noted there for customers to copy and remove mention of private from the sample yaml in favor of this method.

**Review by rbankston — APPROVED** [review]
These changes look great. Like removing the nlb section in favor of specifying it in the documentation

File: charts/nginx/templates/nginx-service.yaml (hunk 1 above)
Comment (danielhoherd) [reply charts/nginx/templates/nginx-service.yaml]: @rbankston if you haven't yet, you may want to turn that into a ticket so the idea doesn't get lost when i resolve this conversation.

**Pull request merged**